From 37132eaf74c52332af7ce99d35143515db32327f Mon Sep 17 00:00:00 2001
From: Tony Layher <layhertony@gmail.com>
Date: Fri, 13 Feb 2015 22:58:19 -0500
Subject: [PATCH] NFCService: Add sysprop to prevent FW download during boot 
 with NFC off.

 * The SEC NFC chip used by samsung for the
   Note4, Note Edge, Alpha and S5-mini does not handle
   the download attempt on boot well if NFC
   is off.
 * If download is attempted while booting
   with NFC off, the NFC device is opened
   and never closed, resulting in a kernel
   wake lock.  It also prevents turning
   NFC on due to the device already being
   open with the service in a tranisition
   from off -> on.
 * Devices can set the property ro.nfc.sec_hal
   to any value to signify the use of the
   sec NFC chip. (S3FWRN)

Change-Id: Ifb28074eb473ab5b7e4b82a14336ee51972d3975
---
 src/com/android/nfc/NfcService.java | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/com/android/nfc/NfcService.java b/src/com/android/nfc/NfcService.java
index 620b0427..41e7e707 100644
--- a/src/com/android/nfc/NfcService.java
+++ b/src/com/android/nfc/NfcService.java
@@ -81,6 +81,7 @@ import android.provider.Settings;
 import android.se.omapi.ISecureElementService;
 import android.service.vr.IVrManager;
 import android.service.vr.IVrStateCallbacks;
+import android.text.TextUtils;
 import android.util.Log;
 
 import com.android.internal.logging.MetricsLogger;
@@ -558,6 +559,15 @@ public class NfcService implements DeviceHostListener {
         }
     }
 
+    boolean isSecHal() {
+        String nfcSecHal = SystemProperties.get("ro.nfc.sec_hal");
+        if (!TextUtils.isEmpty(nfcSecHal)) {
+            Log.i(TAG, "This device uses SEC NFC CHIP.");
+            return true;
+        }
+        return false;
+    }
+
     /**
      * Manages tasks that involve turning on/off the NFC controller.
      * <p/>
@@ -624,7 +634,7 @@ public class NfcService implements DeviceHostListener {
                     if (mPrefs.getBoolean(PREF_NFC_ON, NFC_ON_DEFAULT)) {
                         Log.d(TAG, "NFC is on. Doing normal stuff");
                         enableInternal();
-                    } else {
+                    } else if (!isSecHal()) {
                         Log.d(TAG, "NFC is off.  Checking firmware version");
                         mDeviceHost.checkFirmware();
                     }
-- 
2.17.1

